BrowserKit/Sources/ComponentLibrary/Buttons/CloseButton.swift

```diff
@@ -0,0 +1,40 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Common
+import UIKit
+
+public class CloseButton: UIButton {
+    private var viewModel: CloseButtonViewModel?
+
+    private struct UX {
+        static let closeButtonSize = CGSize(width: 30, height: 30)
+        static let crossCircleImage = StandardImageIdentifiers.ExtraLarge.crossCircleFill
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        setImage(UIImage(named: UX.crossCircleImage), for: .normal)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupConstraints() {
+        NSLayoutConstraint.activate([
+            self.heightAnchor.constraint(equalToConstant: UX.closeButtonSize.height),
+            self.widthAnchor.constraint(equalToConstant: UX.closeButtonSize.width)
```

Comment:
Nit: if the button is going to own the responsibility of applying its own constraints for a fixed width/size then it is probably a good idea for it to also update its `translatesAutoresizingMaskIntoConstraints` to false. Otherwise problems might occur if someone doesn't construct this button subclass using `.build` (or initializes it directly but forgets to set the above property).

Reply:
good call! I think this is a good safety check. We should be using `.build` but its not mandated, so thank you for pointing this out!